src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOpenScmpOperation.cs

```diff
@@ -0,0 +1,176 @@
+
+        /// <summary>
+        /// The error occurred during operation
+        /// </summary>
+        public string ErrorMessage { get; set; }
+
+        // The schema compare public api doesn't currently take a cancellation token so the operation can't be cancelled
+        public void Cancel()
+        {
+        }
+
+        /// <summary>
+        /// Disposes the operation.
+        /// </summary>
+        public void Dispose()
+        {
+            if (!disposed)
+            {
+                this.Cancel();
+                disposed = true;
+            }
+        }
+
+        public void Execute(TaskExecutionMode mode)
+        {
+            if (this.CancellationToken.IsCancellationRequested)
+            {
+                throw new OperationCanceledException(this.CancellationToken);
+            }
+
+            try
+            {
+                SchemaComparison compare = new SchemaComparison(this.Parameters.filePath);
+
+                // load xml file because some parsing still needs to be done
+                this.scmpInfo = new XmlDocument();
+                this.scmpInfo.Load(this.Parameters.filePath);
+
+                this.Result = new SchemaCompareOpenScmpResult()
+                {
+                    DeploymentOptions = new DeploymentOptions(compare.Options),
+                    Success = true,
+                    SourceEndpointInfo = this.GetEndpointInfo(true, compare.Source),
+                    TargetEndpointInfo = this.GetEndpointInfo(false, compare.Target),
+                    OriginalTargetName = this.GetOriginalTargetName(),
+                    OriginalTargetConnectionString = this.GetOriginalTargetConnectionString(),
+                    ExcludedSourceElements = this.GetExcludedElements(compare.ExcludedSourceObjects),
+                    ExcludedTargetElements = this.GetExcludedElements(compare.ExcludedTargetObjects)
+                };
+            }
+            catch (Exception e)
+            {
+                ErrorMessage = e.Message;
+                Logger.Write(TraceEventType.Error, string.Format("Schema compare open scmp operation failed with exception {0}", e.Message));
+                throw;
+            }
+        }
+
+        private SchemaCompareEndpointInfo GetEndpointInfo(bool source, SchemaCompareEndpoint endpoint)
+        {
+            SchemaCompareEndpointInfo endpointInfo = new SchemaCompareEndpointInfo();
+
+            // if the endpoint is a dacpac we don't need to parse the xml
+            SchemaCompareDacpacEndpoint dacpacEndpoint = endpoint as SchemaCompareDacpacEndpoint;
+            if (dacpacEndpoint != null)
+            {
+                endpointInfo.EndpointType = SchemaCompareEndpointType.Dacpac;
+                endpointInfo.PackageFilePath = dacpacEndpoint.FilePath;
+            }
+            else
+            {
+                // need to parse xml to get connection string of database
```

Comment:
@pensivebrian should we make the connection string getter to be public in DacFx SchemaCompareDatabaseEndpoint - so that we don't have to do parsing here? What do you think? I am worried that parsing might need constant maintenance if things around this change in DacFx.

Reply:
I don't have enough context to have an opinion. I'll leave it up to you two.